src/App.tsx

```diff
@@ -9,39 +9,31 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { loadPosts, postsSlice } from './features/posts/postsSlice';
+import { selectedPostSlice } from './features/selectedPost/selectedPostSlice';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const {
+    posts,
+    loaded: loaded,
+    error: hasError,
+  } = useAppSelector(state => state.posts);
 
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const selectedPost = useAppSelector(state => state.selectedPost);
+  const author = useAppSelector(state => state.author);
```

Comment:
The `author` state is being accessed from the Redux store, but it's not clear if this state is defined in the store. Ensure that the `author` state is properly initialized and managed in the Redux store to avoid potential runtime errors.